src/Lumper.UI/ViewModels/Bsp/BspNodeBase.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using DynamicData;
+using DynamicData.Binding;
+using Lumper.UI.Models;
+using ReactiveUI;
+
+namespace Lumper.UI.ViewModels.Bsp;
+
+public abstract class BspNodeBase : ViewModelBase
+{
+    private ReadOnlyObservableCollection<BspNodeBase>? _filteredNodes;
+    private bool _isVisible;
+    private ReadOnlyObservableCollection<BspNodeBase>? _nodes;
+    private BspNodeBase? _selectedNode;
+
+    public BspNodeBase(MainWindowViewModel mainModel)
+    {
+        Parent = null;
+        _isVisible = true;
+        Main = mainModel;
+    }
+
+    public BspNodeBase(BspNodeBase parent)
+    {
+        Parent = parent;
+        _isVisible = parent._isVisible;
+        Main = parent.Main;
+    }
+
+
+    public bool IsModifiedRecursive => IsModified
+                                       || (_nodes is { Count: > 0 } && _nodes.Any(n => n.IsModifiedRecursive));
```

Comment:
80 char line limit would be nice but I didn't do this every time so who am I to judge :) Just noticed it here because there is a newline but it was still very long.

Reply:
Rider by default limits to 120 char's. I will see to add 80 char limit in .editorconfig

Reply:
80 seems to low. I set it to 100 in .editorconfig